source/common/http/conn_manager_utility.h

```diff
@@ -54,13 +63,16 @@ class ConnectionManagerUtility {
    * Note this function may be called twice on the response path if there are
    * 100-Continue headers.
    *
-   * @return the final trusted remote address. This depends on various settings and the
-   *         existence of the x-forwarded-for header. Again see the method for more details.
+   * @return MutateRequestHeadersResult containing the final trusted remote address if detected.
+   *         This depends on various settings and the existence of the x-forwarded-for header.
+   *         Note that an extension might also be used. If detection fails, the result may contain
```

Comment:
don't require either the IP or options for rejecting?
if so s/may/will/ or must

Reply:
Not really, you could fail detection in which case final_remote_address will be nullptr. And, if detection failed that doesn't imply you want rejection. Thus it's really may (not will/must).